packages/wdio-browserstack-service/src/insights-handler.ts

```diff
@@ -0,0 +1,471 @@
+            testData.duration_in_ms = results.duration
+            if (this._hooks[fullTitle]) {
+                testData.hooks = this._hooks[fullTitle]
+            }
+        }
+
+        if (eventType == 'TestRunStarted') {
+            testData['integrations'] = {}
+            if (this._browser && this._platformMeta) {
+                const provider = getCloudProvider(this._browser)
+                testData.integrations[provider] = this.getIntegrationsObject()
+            }
+        }
+
+        let uploadData: UploadType = {
+            event_type: eventType,
+        }
+
+        /* istanbul ignore if */
+        if (eventType.match(/HookRun/)) {
+            testData.hook_type = testData.name?.toLowerCase() ? getHookType(testData.name.toLowerCase()) : 'undefined'
+            uploadData.hook_run = testData
+        } else {
+            uploadData['test_run'] = testData
+        }
+
+        const req = this.requestQueueHandler.add(uploadData)
+
+        if (req.proceed) {
+            await uploadEventData(req.data, req.url)
+        }
+    }
+
+    private async sendTestRunEventForCucumber (world: ITestCaseHookParameter, eventType: string) {
+        const uniqueId = getUniqueIdentifierForCucumber(world)
+
+        let testMetaData = this._tests[uniqueId]
+        if (!testMetaData) testMetaData = {}
+
+        const { feature, scenario, steps } = testMetaData
+
+        let fullNameWithExamples = world.pickle.name
+        const examples = getScenarioExamples(world)
+        if (examples) {
+            fullNameWithExamples = world.pickle.name + ' (' + examples.join(', ')  + ')'
+        }
+
+        /* istanbul ignore next */
+        let testData: TestData = {
+            uuid: testMetaData.uuid,
+            started_at: testMetaData.startedAt,
+            finished_at: testMetaData.finishedAt,
+            type: 'test',
+            body: {
+                lang: 'webdriverio',
+                code: null
+            },
+            name: fullNameWithExamples,
+            scope: fullNameWithExamples,
+            scopes: [feature?.name || ''],
+            identifier: scenario?.name,
+            file_name: feature?.path,
+            vc_filepath: (this._gitConfigPath && feature?.path) ? path.relative(this._gitConfigPath, feature?.path) : undefined,
+            location: feature?.path,
+            framework: this._framework,
+            result: 'pending',
+            meta: {
+                feature: feature,
+                scenario: scenario,
+                steps: steps,
+                examples: examples
+            }
+        }
+
+        if (eventType == 'TestRunStarted') {
+            testData['integrations'] = {}
+            if (this._browser && this._platformMeta) testData['integrations'][getCloudProvider(this._browser)] = this.getIntegrationsObject()
+        }
+
+        /* istanbul ignore if */
+        if (world.result) {
+            let result: string = world.result.status.toLowerCase()
```

Comment:
string should be inferred here by TS
```suggestion
            let result = world.result.status.toLowerCase()
```